Firestore/core/src/firebase/firestore/core/query.cc

```diff
@@ -70,16 +70,30 @@ const FieldPath* Query::InequalityFilterField() const {
   return nullptr;
 }
 
-bool Query::HasArrayContainsFilter() const {
+absl::optional<Operator> Query::GetArrayOps() const {
   for (const auto& filter : filters_) {
     if (filter->IsAFieldFilter()) {
       const auto& relation_filter = static_cast<const FieldFilter&>(*filter);
-      if (relation_filter.op() == Operator::ArrayContains) {
-        return true;
+      if (relation_filter.op() == Operator::ArrayContains ||
```

Comment:
We seem to be making these same sets of comparisons in many places (and differently each time--we're using `absl::c_linear_search` in query_core.mm. It seems like it would be worth pulling out free functions like `IsArrayOperator` and `IsDisjunctiveOperator` so that we can use them uniformly.

I suggest chucking both in an `operator.h`. You might as well declare them `inline` there too:

```c++
inline bool IsArrayOperator(Filter::Operator op) {
  return op == Filter::Operator::ArrayContains || op == Filter::Operator::ArrayContainsAny;
}

// others too
```

Reply:
oh wow, didn't know you could inline functions like that. I am definitely a fan of this though -- it looks so much cleaner!